modules/stars.py

```diff
@@ -20,8 +20,280 @@
+
+    async def remove_starer(self, message_id: int, user_id: int):
+        query = """DELETE FROM starers WHERE msg_id={} AND user_id={}""".format(message_id, user_id)
+        await self.pool.execute(query)
+
+    async def update_entry(self, reactions: int, message_id: int):
+        query = """UPDATE starboard_entries SET stars = {} WHERE msg_id={}""".format(reactions, message_id)
+        await self.pool.execute(query)
+
+    async def remove_entry(self, message_id: int):
+        query = """DELETE FROM starboard_entries WHERE msg_id={}""".format(message_id)
+        await self.pool.execute(query)
+
+    async def clear_starers(self, message_id: int):
+        query = """DELETE FROM starers WHERE msg_id={}""".format(message_id)
+        await self.pool.execute(query)
+
+    def get_formatted_time(self):
+        now = date.now()
+        time = now.strftime("%m/%d/%Y %I:%M %p")
+        return time
+
+    async def handle_star(self, payload: discord.RawReactionActionEvent):
+        time = self.get_formatted_time()
+        entry: StarboardEntry = StarboardEntry(self.pool, payload.message_id)
+        await entry.fetch()
+
+        if str(payload.emoji) != STARBOARD_EMOJI:
+            return
+
+        channel: discord.TextChannel = self.bot.get_channel(payload.channel_id)  # type: ignore
+        message: discord.Message = await channel.fetch_message(payload.message_id)
+
+        reaction = discord.utils.get(message.reactions, emoji=STARBOARD_EMOJI)
+        reaction_count = reaction.count if reaction else 0
+
+        if entry.exists:
+            bot_msg_id = entry.bot_message_id
+
+            query = """SELECT * FROM starers WHERE user_id={} AND msg_id={}""".format(payload.user_id, entry.msg_id)
+
+            starer = await self.pool.fetchrow(query)
+
+            if starer is not None:
+                return
+
+            await self.add_starer(payload.user_id, payload.message_id)
+
+            query = """UPDATE starboard_entries SET stars = starboard_entries.stars + 1 
+                                    WHERE msg_id = {}
+                                """.format(
+                entry.msg_id
+            )
+
+            await self.pool.execute(query)
+
+            bot_channel: discord.TextChannel = await self.bot.fetch_channel(self.starboard_channel_id)  # type: ignore
+            bot_message = await bot_channel.fetch_message(bot_msg_id)
+
+            stars = entry.stars + 1
+            star = self.get_star(stars)
+            await bot_message.edit(content=HEADER_TEMPLATE.format(star, stars, payload.channel_id, payload.channel_id))
+            return
+
+        if not reaction_count >= self.entry_requirement:
+            return
+
+        star = self.get_star(reaction_count)
+
+        embed = discord.Embed(color=STARBOARD_EMBED_COLOR)
+
+        if len(message.attachments) > 0:
+            for attachment in message.attachments:
+                filename = attachment.filename
+                if (
```

Comment:
Just create a tuple of all allowed extensions and pass that to a single .endswith rather than this or chain

Reply:
Also consider if file.is_spoiler()